base/doc/clsguide.tex

```diff
@@ -1181,12 +1181,162 @@ \subsection{Querying localisation}
+   \cs{label}/\cs{ref}.
+
+ \item[\texttt{title}] The content of \cs{@currentlabelname}.  This
+   command is filled beside others by the \pkg{nameref} package and
+   some classes (e.g.~\pkg{memoir}) and typically gives the title of
+   some sectioning command.
+
+ \item[\texttt{target}] The content of \cs{@currentHref}.  This
+   command is normally filled by \pkg{hyperref} and gives the name of
+   the last destination it created.
+
+ \item[\texttt{counter}] The content of \cs{@currentcounter}.  This
+   command contains after a \cs{refstepcounter} the name of the
+   counter.
+
+ \item[\texttt{xpos}, \texttt{ypos}] These properties records the
+   $x$~and $y$ coordinates of a point previously stored with
+   \cs{pdfsavepos}/\cs{savepos}.  E.g.~(if \pkg{bidi} is used it can
+   be necessary to save the position before and after the label):
+   \begin{verbatim}
+     \pdfsavepos 
+     \RefProperty{myposition}{xpos,ypos}%
+     \pdfsavepos
+   \end{verbatim}  
+\end{description} 
+ 
+Class and package authors can define more properties to store other
+values they are interested in.
+\begin{decl}
+  |\NewProperty|\arg{name}\arg{setpoint}\arg{default}\arg{code}\\
+  |\SetProperty|\arg{name}\arg{setpoint}\arg{default}\arg{code}
+\end{decl}
+These commands declare or change a property \meta{name}\footnote{The
+standard set of properties and properties of other package should not
+be changed!}. If a new property is declared within a package it is
+suggested that its name is always structured as follows:
+\meta{package-name}\texttt{/}\meta{property-name}. \meta{setpoint} is
+either |now| or |shipout| and decides if the value is written directly
+or at the next shipout. \meta{default} is used if the property is
+referenced but not yet known.  \meta{code} is the code executed when
+storing the value. For example the \texttt{pagenum} property is
+declare as
+\begin{verbatim}
+\NewProperty{pagenum}{shipout}{0}{\the\value{page}}
+\end{verbatim}
+
+The commands related to properties are offered as a set of CamelCase
+commands for traditional \LaTeXe{} packages (and for use in the
+document preamble if needed) as well as \texttt{expl3} commands for
+modern packages, that use the L3 programming layer of \LaTeX{}. The
+\texttt{expl3} commands and more details can by found in
+\texttt{ltproperties-doc.pdf}.
+
+\subsection{Preparing link targets}
+
+Active links in a document need targets where they can jump to. Such
+targets are often created automatically (if the package \pkg{hyperref}
+is loaded) by the \cs{refstepcounter} command but there are also cases
+where class or package authors need to add a target manually for
+example in unnumbered sectioning commands or in environments. For this
+\LaTeX{} provides the following commands. \emph{Without}
+\pkg{hyperref} they do nothing or insert only a whatsits, \emph{with}
+\pkg{hyperref} they add the necessary targets. Details about the
+behaviour and the arguments of the following commands can by found in
```

Comment:
'behavior' (sigh, but that's the general position)

Reply:
hm, the rest of the file uses behaviour everywhere (with one exception).  So what is the general position ???

Reply:
Generally use American english. In the color package  I used American English in the code and English English in the documentation (but I'm not sure that was the best idea ever).

Reply:
as David said, in general we try to use American English (with a few historical exceptions) and with the exception that if David or Joseph or Chris write documentation their fingers tend to write British English (sometimes).
"behavior" versus "behaviour" is more borderline as far as I'm concerned (in contrast to writing "you change the colour to red by using `\color{red}`" which indeed I think wasn't the best idea ever :-)

So if Joseph spotted "behaviour" change it thoughout, I would say.